experimental/apps-mcp/lib/providers/clitools/invoke_databricks_cli.go

```diff
@@ -0,0 +1,35 @@
+package clitools
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/databricks/cli/experimental/apps-mcp/lib/common"
+	"github.com/databricks/cli/libs/exec"
+)
+
+// InvokeDatabricksCLI runs a Databricks CLI command and returns the output.
+func InvokeDatabricksCLI(ctx context.Context, command string) (string, error) {
+	if command == "" {
+		return "", errors.New("command is required")
+	}
+
+	executor, err := exec.NewCommandExecutor(".")
+	if err != nil {
+		return "", fmt.Errorf("failed to create command executor: %w", err)
+	}
+
+	// GetCLIPath returns the path to the current CLI executable
+	cliPath := common.GetCLIPath()
```

Comment:
Please set these env vars so we can track CLI invocations through the agent https://github.com/databricks/cli/blob/a82f3f6bc57e211d27f704644fd81de9b3e3bafe/cmd/root/user_agent_upstream.go#L12-L13

Reply:
+1 for adding these, but IIRC executor doesn't currently let you set envirnoment variables, so this might be something for a followup PR.

Btw, there might be a major alternative here: it might work to capture the parent-process ID in the server and/or first tool calls.  Based on that parent-process ID you can likely determine the CLI is called by the agent. Then you can also capture CLI invocations that are done directly, without the MCP helper here.

Reply:
Actually I stand corrected after reading https://github.com/databricks/cli/pull/4027: environment support is already within reach.